packages/react-native-bridge/android/react-native-bridge/src/main/java/org/wordpress/mobile/ReactNativeGutenbergBridge/RNReactNativeGutenbergBridgeModule.java

```diff
@@ -554,14 +556,70 @@ private ConnectionStatusCallback requestConnectionStatusCallback(final Callback
         };
     }
 
+    @ReactMethod
+    public void showAndroidSoftKeyboard() {
+        Activity currentActivity = mReactContext.getCurrentActivity();
+        if (isAnyViewFocused()) {
+            // Cancel any previously scheduled Runnable
+            if (mKeyboardRunnable != null) {
+                currentActivity.getWindow().getDecorView().removeCallbacks(mKeyboardRunnable);
+            }
+
+            View currentFocusedView = getCurrentFocusedView();
+            if (currentFocusedView != null) {
+                currentFocusedView.getViewTreeObserver().addOnWindowFocusChangeListener(new ViewTreeObserver.OnWindowFocusChangeListener() {
+                    @Override
+                    public void onWindowFocusChanged(boolean hasFocus) {
+                        if (hasFocus) {
+                            mKeyboardRunnable = createShowKeyboardRunnable(currentActivity);
+                            currentActivity.getWindow().getDecorView().post(mKeyboardRunnable);
+                            currentFocusedView.getViewTreeObserver().removeOnWindowFocusChangeListener(this);
+                        }
+                    }
+                });
+            }
+        }
+    }
+
+    private Runnable createShowKeyboardRunnable(final Activity currentActivity) {
```

Comment:
I noticed that in this function we have the argument `currentActivity` but in the function `getCurrentFocusedView` also used in this code it's not needed, as it's retrieved from the React context. This makes me wonder if we really need the argument. WDYT?

Reply:
We do need the current activity for this line https://github.com/WordPress/gutenberg/pull/57069/commits/b0bfdf10f48440d69d585aeb379c5c9ad9fe4f56#diff-3358b6e4c037a95586241526f5478f87905bd599ac2ffe341525d1a62aa1320fR590 if we remove the argument we'd need to get it from within the `createShowKeyboardRunnable` function, is this what you mean? Just to double check 😄

Reply:
Yeah, I thought that following the implementation of `getCurrentFocusedView` we could get the current activity from the React context. This way we wouldn't need to have the argument. Does it make sense?

https://github.com/WordPress/gutenberg/blob/b0bfdf10f48440d69d585aeb379c5c9ad9fe4f56/packages/react-native-bridge/android/react-native-bridge/src/main/java/org/wordpress/mobile/ReactNativeGutenbergBridge/RNReactNativeGutenbergBridgeModule.java#L603

Alternatively, we could add the `currentActivity` argument to the rest of the recently incorporated functions (`getCurrentFocusedView` and `isAnyViewFocused`), so we only retrieve it in `showAndroidSoftKeyboard`.

Reply:
Ahh I see! I've updated it in https://github.com/WordPress/gutenberg/pull/57069/commits/1db7e475989bcd180b055bd2f61c611c8d1bd91e to get the activity within `createShowKeyboardRunnable`